bin/runtime/src/lib.rs

```diff
@@ -895,6 +896,68 @@ impl pallet_feature_control::Config for Runtime {
     type Supervisor = EnsureRoot<AccountId>;
 }
 
+parameter_types! {
+    pub const DisallowPermissionlessEnterDuration: AlephBlockNumber = 0;
+    pub const DisallowPermissionlessExtendDuration: AlephBlockNumber = 0;
+
+    // Safe mode on enter will last 900 block, which equals to 1 session
+    pub const RootEnterDuration: AlephBlockNumber = 900;
+    // Safe mode on extend will last additional 900 block, which equals to 1 session
+    pub const RootExtendDuration: AlephBlockNumber = 900;
+
+    pub const DisallowPermissionlessEntering: Option<Balance> = None;
+    pub const DisallowPermissionlessExtending: Option<Balance> = None;
+    pub const DisallowPermissionlessRelease: Option<AlephBlockNumber> = None;
+}
+
+/// Calls that can bypass the safe-mode pallet.
+pub struct SafeModeWhitelistedCalls;
+impl Contains<RuntimeCall> for SafeModeWhitelistedCalls {
+    fn contains(call: &RuntimeCall) -> bool {
+        matches!(
+            call,
+            RuntimeCall::System(_) | RuntimeCall::SafeMode(_) | RuntimeCall::Timestamp(_)
```

Comment:
How about `RuntimeCall::Sudo`? Is it allowed by deafult, or we don't want it?

Reply:
added